floating point test failures: test_stats.py and test_otu_significance.py

When running test_stats.py and test_otu_significance.py on the QIIME 1.8.0 AMI that has had numpy 1.7.1 installed via pip (not the version bundled under `$HOME/qiime_software`), I get the following test failures:

```
======================================================================
FAIL: test_call_small (__main__.PartialMantelTests)
Test the running of partial Mantel analysis on small input.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/workspace/qiime-pip-test/qiime/tests/test_stats.py", line 1615, in test_call_small
    p_val_key='mantel_p')
  File "/mnt/workspace/qiime-pip-test/qiime/tests/test_stats.py", line 138, in assertCorrectPValue
    self.assertTrue(found_match)
AssertionError: False is not true
```

and

```
======================================================================
FAIL: test_run_correlation_test (__main__.GroupedCorrelationTests)
Test run_correlation_test works.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/workspace/qiime-pip-test/qiime/tests/test_otu_significance.py", line 861, in test_run_correlation_test
    self.assertFloatEqual(exp_bootstrapped_pvals, obs_pvals)
  File "/mnt/shiningpanda/jobs/1a4e5e78/virtualenvs/da4bc1d6/local/lib/python2.7/site-packages/cogent/util/unit_test.py", line 287, in assertFloatEqual
    self.assertFloatEqualRel(observed, expected, rel_eps)
  File "/mnt/shiningpanda/jobs/1a4e5e78/virtualenvs/da4bc1d6/local/lib/python2.7/site-packages/cogent/util/unit_test.py", line 205, in assertFloatEqualRel
    (`observed`, `expected`, `diff`))
AssertionError: Got 0.546, but expected 0.54700000000000004 (diff was -0.0010000000000000009)
```

The first failure is probably something similar to #1239. I'll fix this one.

The second failure is possibly related to #1305. @wdwvt1 can you look into this failure?

--- a/tests/test_otu_significance.py
+++ b/tests/test_otu_significance.py
@@ -4,7 +4,7 @@ from __future__ import division
 
 __author__ = "Luke Ursell"
 __copyright__ = "Copyright 2013, The QIIME project"
-__credits__ = ["Luke Ursell, Will Van Treuren"]
+__credits__ = ["Luke Ursell", "Will Van Treuren", "Jai Ram Rideout"]
 __license__ = "GPL"
 __version__ = "1.8.0-dev"
 __maintainer__ = "Luke Ursell"
@@ -843,8 +843,7 @@ class GroupedCorrelationTests(TestCase):
                                   0.045999999999999999,
                                   0.38200000000000001,
                                   0.88200000000000001,
-                                  0.438,
-                                  0.54600000000000004]
+                                  0.438]
 
         obs_ccs, obs_pvals = run_correlation_test(data_gen, 'pearson',
                                                   CORRELATION_TEST_CHOICES,
@@ -858,7 +857,17 @@ class GroupedCorrelationTests(TestCase):
         obs_ccs, obs_pvals = run_correlation_test(data_gen, 'pearson',
                                                   CORRELATION_TEST_CHOICES, pval_assignment_method='bootstrapped',
                                                   permutations=1000)
-        self.assertFloatEqual(exp_bootstrapped_pvals, obs_pvals)
+
+        # We can use the default epsilon in assertFloatEqual to test all
+        # p-values but the last one. This last one needs a different epsilon
+        # because the 479th permuted correlation coefficient can differ in
+        # equality to the observed correlation coefficient depending on the
+        # platform/numpy installation/configuration. Thus, the count of
+        # more-extreme correlation coefficients can differ by 1. With 1000
+        # permutations used in the test, this difference is 1 / 1000 = 0.001,
+        # hence the larger epsilon used here.
+        self.assertFloatEqual(obs_pvals[:-1], exp_bootstrapped_pvals)
+        self.assertFloatEqual(obs_pvals[-1], 0.54600000000000004, eps=1e-3)
 
         # spearman
         exp_ccs = [0.25714285714285712,

